crates/pallet-subspace/src/lib.rs

```diff
@@ -241,6 +241,11 @@ mod pallet {
 
         /// Weight information for extrinsics in this pallet.
         type WeightInfo: WeightInfo;
+
+        /// Total number of segments in the blockchain
+        fn total_number_of_segments() -> u64 {
+            RecordsRoot::<Self>::count() as u64
+        }
```

Comment:
I have never seen things added to config like this. It is typically added as a method on a pallet itself instead.

Also prefer `u64::from()` rather than `as u64` in the runtime to make sure it stops compiling if the type of `count()` method changes in incompatible way (lets say to `u128` or `f64` for whatever reason).

Also I feel like API will have better usability if we return number of pieces from here instead of number of segments.